### PR TITLE
[FS-672] Improve swagger docs for Client fields.

### PR DESCRIPTION
**Pull request by stephen-smith**
[FS-672](https://wearezeta.atlassian.net/browse/FS-672)

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] I updated **changelog.d** subsections with one or more entries.

### CODE REVIEW

**Comment by pcapriotti** [comment]
It would be good to also improve the example. Right now it reads:
```
    "mls_public_keys": {
      "additionalProp1": "string",
      "additionalProp2": "string",
      "additionalProp3": "string"
    },
```
which is not very useful.

File: libs/wire-api/src/Wire/API/User/Client.hs [hunk 1]
```diff
@@ -477,6 +477,22 @@ data Client = Client
...
+             . (example ?~ toJSON (Map.fromList $ map (,exampleValue) keys))
+         )
+    & named "MLSPublicKeys"
+  where
+    keys :: [SignatureSchemeTag]
+    keys = [minBound .. maxBound]
+    exampleValue = fromMaybe (toJSON ("base64==" :: Text)) (valueSchema ^. doc . example)
+    valueSchema = base64Schema
+    mapSchema :: ValueSchema SwaggerDoc MLSPublicKeys
+    mapSchema = map_ valueSchema
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/User/Client.hs:494]:
```suggestion

    exampleValue :: A.Value
    exampleValue = fromMaybe (toJSON ("base64==" :: Text)) (valueSchema ^. doc . example)

    mapSchema :: ValueSchema SwaggerDoc MLSPublicKeys
    mapSchema = map_ base64Schema
```

(I think these are questions of taste.)

**Review by fisx — COMMENTED** [review]
when building this, running `http://localhost:8080/swagger-ui/`, and opening http://localhost:8080/swagger-ui/, I see this:

2022-09-01-120754_1920x1080_scrot

`MLSPublicKeys` shouldn't be empty, should it?

**Comment by stephen-smith** [comment]
> `MLSPublicKeys` shouldn't be empty, should it?

It doesn't have any fields, so yes, I think it is supposed to be empty in that display.

As best as I can tell, that's normal for map/dict types in swagger.

The example should have the only valid key and an example value, and there should also be a description if you expand the name.

image

image

**Review by pcapriotti — APPROVED** [review]
LGTM

**Review by fisx — APPROVED** [review]

**Pull request merged**